### PR TITLE
Resolved connection leak in SNI by adding dispose for underlying TCPstream in SNITcpHandle.Dispose()

### PR DESCRIPTION
**Pull request by saurabh500**
Fix for SqlConnection TCP connection leak.

Copied from master branch PR https://github.com/dotnet/corefx/pull/13653
Originally caused by https://github.com/dotnet/corefx/commit/30abd30421289509cd3bd18a9114501afdc0fb7c which removed a TcpClient.Dispose() in SNITcpHandle

This is a fix for https://github.com/dotnet/corefx/issues/13422

### CODE REVIEW

**Comment by karelz** [comment]
Blocked on shiproom approval.

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs [hunk 1]
```diff
@@ -57,6 +57,12 @@ public override void Dispose()
                     _sslStream.Dispose();
                     _sslStream = null;
                 }
+
+                if (_tcpStream != null)
+                {
+                    _tcpStream.Dispose();
```
Comment (stephentoub) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs]: Looks like the ctor stores _tcpStream into _stream.  Do we need to also null out _stream?

Also looks like most access to _tcpStream and _stream is done under the same lock used here in Dispose, but there are a few cases where _stream is set to either _sslStream or _tcpStream not under the lock (or at least not in that method, maybe it's held by a caller)... is that ok?

Comment (corivera) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs]: _stream only ever stores _tcpStream or _sslStream, which are both disposed here (if not disposed already). 

Increased async safety in EnableSsl/DisableSsl can be handled in a separate improvement.

Comment (stephentoub) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs]: Yes, but _stream isn't nulled out, which means code could still end up referencing it, even though both _tcpStream and _sslStream have been disposed and nulled out.  Is that ok?

Comment (corivera) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs]: Added tracking issue for EnableSsl/DisableSsl here: https://github.com/dotnet/corefx/issues/15669

Comment (saurabh500) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs]: @stephentoub 

>Yes, but _stream isn't nulled out, which means code could still end up referencing it, even though both _tcpStream and _sslStream have been disposed and nulled out. Is that ok?

The _stream should be nulled out as well. I will make the change.

Comment (stephentoub) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs]: Thanks. LGTM.

**Pull request merged**